### PR TITLE
chore: refurbish Python code by applying refurb linter rules

### PR DESCRIPTION
**Pull request by bowenliang123**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

- polish Python code by applying refurb linter rule: https://docs.astral.sh/ruff/rules/#refurb-furb
  - `refurb` is a tool for improving Python code for better practise : https://github.com/dosisod/refurb
 - no behaviour changes, some notable improvement:
   -  prefer `Path(...).read_bytes()` and `Path(...).read_text()` to read whole file, avoiding manually handling file opennings
   -  prefer `Path(...).write_bytes()`  to write whole file in bytes, avoiding manually handling file opennings

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @laipz8200

File: api/tests/unit_tests/core/app/segments/test_factory.py [hunk 1]
```diff
@@ -29,7 +30,7 @@ def test_integer_variable():
 
 
 def test_float_variable():
-    test_data = {"value_type": "number", "name": "test_float", "value": 3.14}
+    test_data = {"value_type": "number", "name": "test_float", "value": math.pi}
```
Comment (laipz8200) [inline api/tests/unit_tests/core/app/segments/test_factory.py]: I don’t think it makes sense xD

**Review by laipz8200 — CHANGES_REQUESTED** [review]
Almost LGTM, but a few changes still need to be checked.

File: api/tests/unit_tests/core/app/segments/test_factory.py (hunk 1 above)
Comment (bowenliang123) [reply api/tests/unit_tests/core/app/segments/test_factory.py]: Fixed, by ignoring the FURB152  math-constant rule.

**Review by laipz8200 — APPROVED** [review]
LGTM

**Pull request merged**